sambacc/commands/ctdb.py

```diff
@@ -261,12 +261,48 @@ def ctdb_migrate(ctx: Context) -> None:
         ctdb.archive_tdb(ctx.instance_config, ctx.cli.archive)
 
 
-def _lookup_hostname(hostname):
-    # XXX this is a nasty little hack.
-    ips = socket.gethostbyname_ex(hostname)[2]
-    addr = [ip for ip in ips if ip != "127.0.0.1"][0]
-    _logger.info(f"Determined address for {hostname}: {addr}")
-    return addr
+def _lookup_hostname(hostname: str) -> str:
+    try:
+        addrinfo = socket.getaddrinfo(
+            hostname, None, family=socket.AF_UNSPEC, type=socket.SOCK_STREAM
+            )
+        ipv4_addresses = []
+        ipv6_addresses = []
```

Comment:
We are only interested in the first address. There's no need to keep a list.

Reply:
yeah that makes sense to avoid keeping track of list!